### PR TITLE
interviews: add a doc on the internship program

### PR DESCRIPTION
**Pull request by nicks**
Hello @maiamcc, @dbentley,

Please review the following commits I made in branch nicks/intern:

1427c537d31e237782813bb65fc0ff1723258073 (2021-01-19 17:54:22 -0500)
interviews: add a doc on the internship program

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: interviews/internships.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  learn about what they do day-to-day.
+
+- Conversely, everyone on the team should have the opportunity to pair with the
+  intern and help them grow. The ability to learn and teach is an
+  important part of being an engineer.
+  
+## Non-Goals
+
+- At some companies, a successful internship can turn into a full-time offer.
+  At Tilt, this is a non-goal. Small startup hiring is highly contrained, and
```
Comment (maiamcc) [inline interviews/internships.md:23]:
```suggestion
  At Tilt, this is a non-goal. Small startup hiring is highly constrained, and
```

File: interviews/internships.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+- Prior job experience: This might be a previous internship, or a previous
+  job. An internship at Tilt is not a good first internship.
+
+- Prior project experience: This might be experience working with Go (or a
+  Go-like language), or experience working in a microservice-based architecture.
+  Having something familiar to anchor against can help with feeling overwhelmed,
+  since the intern will be learning a lot in a short time!
+
+- Autonomy/Resoluteness: A school project has a lot of structure (a syllabus,
+  deadlines, clear instructions, known solutions, etc.).  An engineering project
```
Comment (maiamcc) [inline interviews/internships.md:58]:
```suggestion
  deadlines, clear instructions, known solutions, etc.). An engineering project
```

File: interviews/internships.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  at a large company has less structure, but still might have multi-year
+  roadmaps. At a small startup, you might not know what you're working on next
+  month. This is not a good internship for a student who's looking for a lot of
+  predictability. So we look for demonstrated ability to work independently and
+  push through failed attempts.
```
Comment (maiamcc) [inline interviews/internships.md:63]: might be good to clarify that this bullet point is not about the ability to do it all yourself, but the ability to push the project forward even via (especially via) asking for help and pairing -- is that a good summary? (B/c that was the point I was most unsure about.)

**Review by maiamcc — APPROVED** [review]
V helpful, ty!

File: interviews/internships.md (hunk 1 above)
Comment (nicks) [reply interviews/internships.md:23]: :grimacing:

File: interviews/internships.md (hunk 2 above)
Comment (nicks) [reply interviews/internships.md]: done

File: interviews/internships.md (hunk 3 above)
Comment (nicks) [reply interviews/internships.md]: i really like this wording! done

**Pull request merged**